File: BHAVRP_Java/BHAVRPv2.0/BHAVRP2.0/src/cujae/inf/ic/om/assignment/classical/Heuristic.java
package cujae.inf.ic.om.assignment.classical;

import java.util.ArrayList;

import cujae.inf.ic.om.assignment.Assignment;
import cujae.inf.ic.om.problem.input.Problem;
import cujae.inf.ic.om.problem.solution.Cluster;

public abstract class Heuristic extends Assignment {
	
	/**
	 * @param  void
	 * @return ArrayList<Cluster> listado de clusters
	 * Inicializa los clusters de la solución con los identificadores de los depósitos
	 **/
	protected ArrayList<Cluster> initializeClusters(){
		ArrayList<Cluster> listClusters = new ArrayList<Cluster>();

		Cluster cluster;
		ArrayList<Integer> listIDItems;

		int totalClusters = Problem.getProblem().getDepots().size();

		for(int i = 0; i < totalClusters; i++)
		{
			listIDItems = new ArrayList<Integer>();
			cluster = new Cluster(Problem.getProblem().getListIDDepots().get(i).intValue(), 0.0, listIDItems); 
			listClusters.add(cluster);
		}
		
		System.out.println("--------------------------------------------------");
		System.out.println("LISTA DE CLUSTERS");
		for(int i = 0; i < listClusters.size(); i++)
		{
			System.out.println("ID CLUSTER: " + listClusters.get(i).getIDCluster());
			System.out.println("DEMANDA DEL CLUSTER: " + listClusters.get(i).getRequestCluster());
			System.out.println("ELEMENTOS DEL CLUSTER: " + listClusters.get(i).getItemsOfCluster());
		}
		System.out.println("--------------------------------------------------");

		return listClusters;
	}

}

File: BHAVRP_Python/cujae/inf/citi/om/heuristic/assignment/classical/heuristic.py
from typing import List
from ..assignment import Assignment
from ....problem.input.problem import Problem
from ....problem.output.solution.cluster import Cluster

class Heuristic(Assignment):
    
    # Método para inicializar una lista de clusters.
    def initialize_clusters(self) -> List[Cluster]:
        list_clusters: List[Cluster] = []
        
        total_clusters = len(Problem.get_problem().get_depots())
        
        for i in range(total_clusters):
            list_id_items: List[int] = []
            cluster = Cluster(Problem.get_problem().get_list_id_depots()[i], 0.0, list_id_items)
            list_clusters.append(cluster)
        
        print("--------------------------------------------------")
        print("LISTA DE CLUSTERS")
        
        for cluster in list_clusters:
            print(f"ID CLUSTER: {cluster.get_id_cluster()}")
            print(f"DEMANDA DEL CLUSTER: {cluster.get_request_cluster()}")
            print(f"ELEMENTOS DEL CLUSTER: {cluster.get_items_of_cluster()}")
        print("--------------------------------------------------")

        return list_clusters